doc/contributing/adding-v8-fast-api.md

```diff
@@ -0,0 +1,153 @@
+# Adding V8 Fast API
+
+Node.js uses [V8](https://github.com/v8/v8) as the JavaScript engine.
+In order to provide fast paths for functions that are called quite often,
+V8 offers the usage of `fast API` functions that does not trigger JavaScript
+functions or allocate heap on JavaScript.
```

Comment:
Modified to be a bit higher level on what the point of fast calls is. this exact wording doesn't have to be used

```suggestion
Node.js uses [V8](https://v8.dev/) as its JavaScript engine.
Embedder functions implemented in C++ incur a high overhead, so V8
provides an API to implement fast-path C functions which may be invoked directly from
JITted code. These functions also come with additional constraints, for example, they
may not trigger garbage collection.
```

Reply:
Thanks! Updating.